packages/typespec-ts/src/modular/serialization/serializers.ts

```diff
@@ -292,14 +293,12 @@ function getPropertyDeclaration(
 function getPropertyInitializer(
   options: SerializeTypeOptions<SdkModelPropertyType>
 ): string {
-  const { type, valueExpr } = options;
+  const { type } = options;
 
   const propSerializer = serializeType({
     ...options,
     type: { ...type.type, nullable: false, optional: false }
   });
 
-  return type.nullable
-    ? `((${valueExpr}) === null) ? (${valueExpr}) : (${propSerializer})`
-    : propSerializer;
```

Comment:
not sure if this is the right fix for null serializer in Modular. @joheredi @dgetu